src/core/app.lua

```diff
@@ -119,66 +122,134 @@ end
 -- new app network by making the changes needed.
 function configure (new_config)
    local actions = compute_config_actions(configuration, new_config)
-   apply_config_actions(actions, new_config)
-   configuration = new_config
```

Comment:
The code in this PR as-is introduces no race conditions AFAIU; do you agree?

As far as whether the leader/follower setup from https://github.com/Igalia/snabb/blob/yang/src/apps/config/README.md introduces race conditions: the follower does not read from `app.configuration` at all.  The leader has an idea of the app graph of the followers because it knows the current high-level configuration of the instance as a whole, and it can get the app graph (the thing you get when you call `core.config.new()`; there are two different things here with the name "configuration" and I am calling `core.config` the app graph to differentiate) via `local app_graph = setup_fn(conf)`.  When a new configuration comes, the leader computes a new app graph, calls `compute_config_actions` to get the config actions, then ships those actions to all followers.

The followers can lag the leader of course.  If there is an operation that for some reason requires that the follower state matches the leader state, the leader can just wait until the followers have consumed all their updates.

Hope that helps.  Do you see a race condition there?  I don't but I would be delighted if you had a counterexample.